elastic/kibana
# [TableListView] Refactor to function component

This PR is part of the issue to refactor the `<TableListView />` component. (https://github.com/elastic/kibana/issues/135020). I decided to split the work in smaller PRs to make it easier to review.

In this initial PR the changes are: converting the previous Class component to a Function component. I've decided to have a reducer to manage the state as it is the best way to make multiple changes to the component state at once and we expect the state management of the component to grow in complexity with future work on the UX redesign.

## How to test

Navigate to the dashboard list view and make sure that everything works as expected

* List the dashboard
* Query (search bar)
* Delete one or multiple dashboard
* Tagging column appear if there are tags
* Edit a dashboard

**Note:** This PR is going to the feature branch. Tests will be added later (https://github.com/elastic/kibana/issues/135811). This PR is _just_ converting the actual component without adding any new functionalities.

**Note 2:** There is an issue in EUI with the sorting controlled by the consumer. I've opened an issue (https://github.com/elastic/eui/issues/6032). Until it is fixed, the "updatedAt" column can't toggle between "asc" and "desc".
